base/download.jl

```diff
@@ -35,8 +35,14 @@ function find_curl()
     end
 end
 
-function download_curl(curl_exe, url, filename)
-    run(`$curl_exe -s -S -g -L -f -o $filename $url`)
+function download_curl(curl_exe::AbstractString, url::AbstractString, filename::AbstractString)
+    err = PipeBuffer()
+    process = run(pipeline(`$curl_exe -s -S -g -L -f -o $filename $url`, stderr=err), wait=false)
```

Comment:
drop `wait=false` and `wait(process)`

Reply:
We can't since with `wait=true`   `run` actually  throws a failed process error exception. 

```
  run(command, args...; wait::Bool = true)

  Run a command object, constructed with backticks. Throws an error if anything goes wrong, including
  the process exiting with a non-zero status (when wait is true).
```

Reply:
@vtjnash by replacing    the error check with `success`, as in your next suggestion, we can drop the `wait` condition, since `success` does that for you